addons/account/models/account_payment_method.py

```diff
@@ -152,7 +152,7 @@ def unlink(self):
     def write(self, vals):
         if 'payment_account_id' in vals:
             account = self.env['account.account'].browse(vals['payment_account_id'])
-            if not account.reconcile:
+            if not account.reconcile and account.internal_type != 'liquidity' and account.internal_group != 'off_balance':
```

Comment:
Why is it not done for the `create`

Instead of sopy/pasting things in `write` and `create`, why not doing something like

```python
def _sanityze_vals(self, vals):
    if 'payment_account_id' in vals:
        account = self.env['account.account'].browse(vals['payment_account_id'])
        if not account.reconcile and account.internal_type != 'liquidity' and account.internal_group != 'off_balance':
            account.reconcile = True
        return vals

def create(self, vals_list):  # api.model_create_multi
    return super().create([self._sanityze_vals(vals) for vals in vals_list])

def write(self, vals):
    return super().write(self._sanityze_vals(vals)
```

Reply:
By the way, there is a bug here
https://github.com/odoo/odoo/blob/59efe7f8c862cb26d9914413e5d776cc8bfab6d9/addons/account/models/account_payment_method.py#L161

Reply:
Agreed, except it's spelled "sanitize" :p

Reply:
"Sanitize" is so 2018
https://www.crosstex.com/sanityzetm-waterless-moisturizing-antimicrobial-gel-116